app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
+  end
+
+
+  def has_company?
+    has_membership_application? && (membership_applications.select { |app| app.company && app.company.persisted? }).count > 0
+  end
+
+
+  def membership_application
+    has_membership_application? ? membership_applications.last : nil
+  end
+
+
+  def company
+    has_company? ? membership_application.company : nil
+  end
```

Comment:
This is checking to see if the user is associated with an instantiated company, via their membership_application
It is possible that the company has not been instantiated. This is the state when a user has submitted their membership_application, but their application has not been approved.
IOW:
- User submits a membership_application.
       -->   membership_application.company == nil
         --> user.company == nil
   
- A SHF admin approves the user's membership_application.  This will set the membership_application.company to an instantiated Company (see `controller/membership_application#accept_application`
     --> membership_application.company == Company.find_or_create(<based on the company_number>)
       --> user.company == <a Company>
       
(FYI - One of the items in PivotalTracker is to use a state machine to handle this and the additional states that will be implemented in the future: https://www.pivotaltracker.com/story/show/134267809)

A user might have many membership_applications if they submit one and it's rejected.  It's not yet clear if they would always edit their previously submitted-and-rejected application, or if they would create a new application.  We're going with the _has_many_ association right now to allow for the possibility of a user submitting many applications.
You're right -- it may be that a user can be associated with (employed by) many companies.  That's also not yet clear. :-)